lib/AST/ASTPrinter.cpp

```diff
@@ -6307,14 +6308,25 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
   }
 
   void visitOpenedArchetypeType(OpenedArchetypeType *T) {
-    if (auto parent = T->getParent()) {
-      printArchetypeCommon(T);
-      return;
-    }
+    if (Options.PrintForSIL) {
+      Printer << "@opened(\"" << T->getOpenedExistentialID() << "\", ";
+      visit(T->getGenericEnvironment()->getOpenedExistentialType());
+      Printer << ") ";
+
+      llvm::DenseMap<CanType, Identifier> newAlternativeTypeNames;
```

Comment:
Seems like this DenseMap could be a SmallDenseMap since it doesn't escape, normally has one entry, and we regularly print giant .sil files (like up to 100MB)